dev-tools/packaging/packages.yml

```diff
@@ -1,9 +1,199 @@
 ---
 
 # This file contains the package specifications for both Community Beats and
-# Official Beats. The shared section contains YAML anchors that are used to
-# define common parts of the package in order to not repeat ourselves.
+# Official Beats.
 
+# List all the available platforms
+platforms: &all-platforms
+  - &linux-amd64
+    os: linux
+    arch: x86_64
+  - &linux-arm64
+    os: linux
+    arch: arm64
+  - &windows-amd64
+    os: windows
+    arch: x86_64
+  - &darwin-amd64
+    os: darwin
+    arch: x86_64
+  - &darwin-arm64
+    os: darwin
+    arch: aarch64
+
+# List all the package type constants (see dev-tools/mage/pkgcommon/pkgcommon-types.go)
+packageTypes: &all-package-types
+  - &pkg-type-rpm
+      1 # RPM
+  - &pkg-type-deb
+      2 # Deb
+  - &pkg-type-zip
+      3 # zip
+  - &pkg-type-targz
+      4 # tar.gz
+  - &pkg-type-docker
+      5 # docker
+
+# Settings section contains general compiling and packaging settings
+settings:
+  fips:
+    compile:
+      cgo: true
+      env:
+        GOEXPERIMENT: systemcrypto
+      tags:
+        - requirefips
+      platforms:
+        - *linux-amd64
+        - *linux-arm64
+
+# List *all* the components available for packaging in elastic-agent
+components:
+  # general template for new components
+  #  - &comp-<anchor name>
+  #    projectName: <project name under manifest>
+  #    packageName: <template representing the name of the archive to download/extract>
+  #    binaryName: <binary name>
+  #    rootDir: <template representing directory from the archive root where the binary is located>
+  #    fips: <boolean indicating if this components whether FIPS compliant>
+  #    platforms:
+  #      - <platform, use anchors for platforms above>
+  #    pythonWheel: true # true if it's a package that is not platform specific
+  #    packageTypes: *all-package-types # package types it should be included in
```

Comment:
nit: I would add what is optional and what isn't here, because I had the impression that rootDir is not optional but in the code we do handle this being empty, so it is optional?! 🙂

Reply:
so this isn't optional then, right? 🙂

Reply:
removed fallback in 9335dc85b651c90008ef59f245d96992a24b7574

Reply:
I don't think that the `fips` attribute is mandatory, or at least I can't see `fips: false` somewhere 🙂

Reply:
added a fixup ;)